tests/model_forms/test_modelchoicefield.py

```diff
@@ -341,6 +342,39 @@ class CustomModelMultipleChoiceField(forms.ModelMultipleChoiceField):
 </div>""" % (self.c1.pk, self.c2.pk, self.c3.pk),
         )
 
+    def test_model_choice_iterator_value_is_hashable(self):
+        model_choice_iterator_value = ModelChoiceIteratorValue(self.c1.pk, self.c1)
+        is_hashable = isinstance(model_choice_iterator_value, collections.Hashable)
+        self.assertTrue(is_hashable)
+
+    def test_model_choice_iterator_value_can_be_used_on_hash_tables(self):
+        categories_with_slug = {self.c1.pk}
+
+        class CustomCheckboxSelectMultiple(CheckboxSelectMultiple):
+            def create_option(self, name, value, label, selected, index, subindex=None, attrs=None):
+                option = super().create_option(name, value, label, selected, index, subindex, attrs)
+                # Modify the HTML based on the object being rendered.
+                if value in categories_with_slug:
+                    option['attrs']['data-slug'] = value.instance.slug
+                return option
+
+        class CustomModelMultipleChoiceField(forms.ModelMultipleChoiceField):
+            widget = CustomCheckboxSelectMultiple
+
+        field = CustomModelMultipleChoiceField(Category.objects.all())
+        self.assertHTMLEqual(
+            field.widget.render('name', []), (
+                '<div>'
+                '<div><label><input type="checkbox" name="name" value="%d" '
+                'data-slug="entertainment">Entertainment</label></div>'
+                '<div><label><input type="checkbox" name="name" value="%d" '
+                '>A test</label></div>'
+                '<div><label><input type="checkbox" name="name" value="%d" '
+                '>Third</label></div>'
+                '</div>'
+            ) % (self.c1.pk, self.c2.pk, self.c3.pk),
+        )
```

Comment:
`test_choice_value_hash` is sufficient. IMO we can remove this test.

Reply:
I agree. I was not sure if I should include this or not.